references/cli/revision.go

```diff
@@ -109,3 +119,132 @@ func NewRevisionListCommand(c common.Args) *cobra.Command {
 	addNamespaceAndEnvArg(cmd)
 	return cmd
 }
+
+// NewRevisionGetCommand gets specific revision of application
+func NewRevisionGetCommand(c common.Args) *cobra.Command {
+	var outputFormat string
+	ctx := context.Background()
+	cmd := &cobra.Command{
+		Use:     "get",
+		Aliases: []string{"get"},
+		Short:   "get specific revision of application",
+		Long:    "get specific revision of application",
+		Args:    cobra.ExactValidArgs(1),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			namespace, err := GetFlagNamespaceOrEnv(cmd, c)
+			if err != nil {
+				return err
+			}
+			name := args[0]
+			def, err := cmd.Flags().GetString("definition")
+			if err != nil {
+				return err
+			}
+
+			return getRevision(ctx, c, outputFormat, cmd.OutOrStdout(), name, namespace, def)
+		},
+	}
+	addNamespaceAndEnvArg(cmd)
+	cmd.Flags().StringP("definition", "d", "", "component definition")
+	cmd.Flags().StringVarP(&outputFormat, "output", "o", "", "raw Application output format. One of: (json, yaml, jsonpath)")
+	return cmd
+}
+
+func getRevision(ctx context.Context, c common.Args, format string, out io.Writer, name string, namespace string, def string) error {
+
+	kubeConfig, err := c.GetConfig()
+	if err != nil {
+		return err
+	}
+	cli, err := c.GetClient()
+	if err != nil {
+		return err
+	}
+
+	dm, err := c.GetDiscoveryMapper()
+	if err != nil {
+		return err
+	}
+
+	pd, err := c.GetPackageDiscover()
+	if err != nil {
+		return err
+	}
+
+	params := map[string]string{
+		"name":      name,
+		"namespace": namespace,
+	}
+	query, err := velaql.ParseVelaQL(MakeVelaQL(revisionView, params, "status"))
```

Comment:
Is there any special reason for using VelaQL for getting application revision here?

It should work if you simple call `cli.Get(ctx, apprev)`. The VelaQL view here is over simple and direct. I would suggest using the client call unless there is any reason for calling VelaQL to get application revision in other cases.

Reply:
My implementation follows the idea from original post. But indeed it seems to a little over by using VelaQL.

Reply:
/cc @wonderflow What do you think?

Reply:
I like the idea to use velaql but it seems not that native. Can we not only get revision from velaql but also parse in velaql? The cli only be a syntax sugar of velaql instead of containing extra logic. If we can't we'd better not abuse velaql in such simple case, it seems over kill.

Reply:
> 

So basically just wrap "vela ql ....." command,  right ?

Reply:
Would that possible for now？@cezhang

Reply:
In my implementation, 
(1) display ui table if no format specified(similar to using vela revision list command), like following
NAME  PUBLISH_VERSION  SUCCEEDED  HASH  BEGIN_TIME  STATUS SIZE
(2) support yaml, json, jsonpath 

if only wrap the "vela ql" command, seems not possible to support feature (1) and jsonpath

Reply:
well, let's leave it as it is for now.